src/dataflow/mod.rs

```diff
@@ -119,8 +119,28 @@ impl Dataflow {
     ///       Uniquely identifies this step for recovery.
     ///   output (bytewax.outputs.Output):
     ///       Output definition.
-    fn output(&mut self, step_id: StepId, output: Output) {
-        self.steps.push(Step::Output { step_id, output });
+    ///   min_batch_size (int):
+    ///       minimum number of messages to receive before writing
+    ///       to the output (default = 1)
+    ///   timeout (Optional[timedelta]):
+    ///       timeout after which items are emitted even if
+    ///       `min_batch_size` is not reached.
+    ///       If this is not set, items won't be emitted until
+    ///       `min_batch_size` is reached.
+    #[pyo3(signature = (step_id, output, min_batch_size = 1, timeout = None))]
+    fn output(
```

Comment:
Ok, as shown in the issue you reference too, it probably makes sense to allow batching anywhere in the dataflow rather than only on output or as a special case behavior of some specific operators.
I'm going to close this PR and open a new one then, so we can discuss the possible problems there.